### DISCUSSION TITLE
USB printers - 'avahi' dependency

### DISCUSSION THREAD

**Discussion by qo0op** [OpenPrinting/ipp-usb#37] [General]
Hi,

I was asking about [Lightweight printing from Linux based embedded system](https://github.com/OpenPrinting/cups/discussions/307) already. This time I would like to ask about ipp-usb 'avahi' dependency.

It's stated, that
- libavahi-common and libavahi-client for DNS-SD
- Running Avahi daemon
are required.

In case that I would like to have solution for "driverless printers" connected only over USB (not needed to share them on the network), is there any possibility to to get rid of 'avahi' dependency?

Thanks.

**Comment by alexpevzner** [comment]
Hi,

even on the USB-only system without network, Avahi is used to notify CUPS that printer is available, and which port (on the loopback interface) it is connected to.

In theory I would be possible to create an Avahi-less build assuming manual configuration of CUPS. But it will add a noticeable maintenance effort, to maintain 2 variants of the package, and taking in account its very limited usefulness, I would prefer not doing it.

Avahi is very lightweight both in terms of CPU usage and disk usage, so I suggest to stay with Avahi.

**Reply by qo0op** [reply]
Ok, thank you for the insight!

**Reply by qo0op** [reply]
Hi @alexpevzner ,

one additional question regarding "lightweight printing with USB printers"..

I'm using cups 2.3.3 ([recipes-extended/cups/cups.inc](http://cgit.openembedded.org/openembedded-core/tree/meta/recipes-extended/cups/cups.inc?h=hardknott)) with ipp-usb daemon. In recipe you can see list of switches in 'EXTRA_OECONF'.

I was thinking about using these additional ones
```
EXTRA_OECONF = " \
...
               --enable-webif=no \
               --disable-browsing \
               --with-local-protocols=no \
...
               "
```

My thinking is that
- I don't need and I don't want to have a web interface on embedded system (it would be good to strip it off fully if possible)
- If I want to use just "local" USB printer(s), not shared on local network
  -> I can disable browsing (hope that I understand it correctly and it means that "external" [printers connected on the same local network won't be browsed](https://www.cups.org/doc/man-cupsd.conf.html))
  -> I don't want to broadcast nothing about my USB printer on local network

I know that this question would fit more to cups discussion, but I want to know your opinion.. If it makes sense or if you have some other optimization suggestions..

Thanks!
